catroid/src/main/java/org/catrobat/catroid/utils/Utils.java

```diff
@@ -479,6 +476,15 @@ public static double round(double value, int precision) {
 		return (double) Math.round(value * scale) / scale;
 	}
 
+	private static void displayDialog(Context context, final Builder builder) {
+		((Activity) context).runOnUiThread(new Runnable() {
+			public void run() {
+				Dialog errorDialog = builder.create();
```

Comment:
Instead of creating the Dialog on the UiThread, pass a Dialog object instead of a Builder object.
If you want to pass in a Builder, just overload the `displayDialog` function :wink: